ambry-router/src/main/java/com.github.ambry.router/GetBlobInfoOperation.java

```diff
@@ -60,19 +60,31 @@
    * @param blobIdStr the blob id associated with the operation in string form.
    * @param futureResult the future that will contain the result of the operation.
    * @param callback the callback that is to be called when the operation completes.
+   * @param operationCompleteCallback the {@link OperationCompleteCallback} to use to complete operations.
    * @param time the Time instance to use.
    * @throws RouterException if there is an error with any of the parameters, such as an invalid blob id.
    */
   GetBlobInfoOperation(RouterConfig routerConfig, ClusterMap clusterMap, ResponseHandler responseHandler,
-      String blobIdStr, FutureResult<BlobInfo> futureResult, Callback<BlobInfo> callback, Time time)
+      String blobIdStr, FutureResult<BlobInfo> futureResult, Callback<BlobInfo> callback,
+      OperationCompleteCallback operationCompleteCallback, Time time)
       throws RouterException {
     super(routerConfig, clusterMap, responseHandler, blobIdStr, futureResult, callback, time);
+    this.operationCompleteCallback = operationCompleteCallback;
     operationTracker = new SimpleOperationTracker(routerConfig.routerDatacenterName, blobId.getPartition(),
         routerConfig.routerGetCrossDcEnabled, routerConfig.routerGetSuccessTarget,
         routerConfig.routerGetRequestParallelism);
   }
 
   /**
+   * {@inheritDoc}
+   */
+  @Override
+  void abort(Exception abortCause) {
+    operationCompleteCallback.completeOperation(operationFuture, operationCallback, null, abortCause);
+    operationCompleted = true;
```

Comment:
Yeah, the ordering does not really matter for this class as at the time this variable is set, no other threads check for its status. This was mainly done for `GetBlobOperation` class because there are multiple threads involved there.